torch/_inductor/scheduler.py

```diff
@@ -413,6 +413,21 @@ def allocate(self):
             V.graph.wrapper_code.codegen_allocation(self.node)
             return
 
+        # Hacky check for if V.kernel using local buffer.
+        # If so, early return to avoid allocation of this buffer
+        # since its usage has been replaced with local buffer.
```

Comment:
Should this go in `remove_kernel_local_buffers`?

Reply:
I feel we need an abstraction to avoid the local buffer usages from contaminating the global stages, e.g., buffer allocation visible globally.